### PR TITLE
Version 0.5 of the app

### PR DESCRIPTION
**Pull request by LaMonF**
Note : Since the branch is very old and many changes result in lot of conflicts with the dev branch, we decided that in this case this branch will be merge to master directly

Changelog : 
- integration of the Geolocation feature
- Geolocation page for each synapse using leaftlet 
- Upgrade all librairies 
- Change voice recorder --> New dedicated page 
- Integration of userfriendly features likes copy pasting on the chat page
- integration of the mute option to the new Kallliope Core api
- CI

### CODE REVIEW

File: kalliope/package.json [hunk 1]
```diff
@@ -1,68 +1,110 @@
...
-    "homepage": "https://kalliope-project.github.io/",
-    "private": true,
-    "scripts": {
-        "clean": "ionic-app-scripts clean",
-        "build": "ionic-app-scripts build",
-        "lint": "ionic-app-scripts lint",
-        "ionic:build": "ionic-app-scripts build",
-        "ionic:serve": "ionic-app-scripts serve"
+  "name": "kalliope",
+  "version": "0.0.1",
```
Comment (Sispheor) [inline kalliope/package.json:3]: 0.5?

File: kalliope/src/pages/synapses/synapses.component.ts [hunk 2]
```diff
@@ -25,43 +35,153 @@ export class SynapsesPage {
...
+                private localNotifications: LocalNotifications,
+                private synapseService: SynapsesService) {
 
+        // load settings from storage
         this.settings = settingsService.getDefaultSettings();
     }
 
     ngOnInit() {
-        this.getSynapse();
+        if (this.settings == null) {
```
Comment (Sispheor) [inline kalliope/src/pages/synapses/synapses.component.ts:49]: this part appears a lot of time. Could be refactored in a Utils method and callable from every component?

File: kalliope/src/pages/synapses/synapses.component.ts [hunk 3]
```diff
@@ -25,43 +35,153 @@ export class SynapsesPage {
...
+            console.log("[SynapsesPage] Settings not loaded. Redirect to settings page");
+            this.navCtrl.push(SettingsPage);
+        } else {
+            console.log("[SynapsesPage] Settings loaded. Url: " + this.settings.url);
+            this.getSynapsesToDisplay();
+        }
+    }
+
+    private geofenceSubscription() {
```
Comment (Sispheor) [inline kalliope/src/pages/synapses/synapses.component.ts:58]: missing doc

File: kalliope/package.json (hunk 1 above)
Comment (LaMonF) [reply kalliope/package.json:3]: Should we align the version number with Kalliope Core ?

Comment (Sispheor) [reply kalliope/package.json:3]: It's just that the PR is named v5...

**Pull request merged**